Source/WebCore/page/DragController.cpp

```diff
@@ -832,47 +832,47 @@ Element* DragController::draggableElement(const LocalFrame* sourceFrame, Element
     if (ImageOverlay::isOverlayText(startElement))
         return selectionDragElement.get();
 
-    for (auto* element = startElement; element; element = element->parentOrShadowHostElement()) {
+    for (RefPtr element = startElement; element; element = element->parentOrShadowHostElement()) {
         auto* renderer = element->renderer();
         if (!renderer)
             continue;
 
         UserDrag dragMode = renderer->style().userDrag();
         if (m_dragSourceAction.contains(DragSourceAction::DHTML) && dragMode == UserDrag::Element) {
             state.type = DragSourceAction::DHTML;
-            return element;
+            return element.get();
```

Comment:
All return statements have a RefPtr now. As a result, we can update this function to return a `RefPtr` without any ref-counting churn. The return statement would `return element;` instead of `return element.get()`.

This will actually be more efficient if the call sites of this functions need a RefPtr.

Reply:
Nice, done.